hapi-fees/src/main/java/com/hedera/services/usage/file/FileOpsUsage.java

```diff
@@ -65,6 +69,19 @@ static int bytesInBaseRepr() {
 	public FileOpsUsage() {
 	}
 
+	public void fileAppendUsage(
+			SigUsage sigUsage,
+			FileAppendMeta appendMeta,
+			BaseTransactionMeta baseMeta,
+			UsageAccumulator accumulator
+	) {
+		accumulator.resetForTransaction(baseMeta, sigUsage);
+
+		final var bytesAdded = appendMeta.getBytesAdded();
+		accumulator.addBpt(LONG_BASIC_ENTITY_ID_SIZE + bytesAdded);
+		accumulator.addSbs(bytesAdded * appendMeta.getLifetime());
```

Comment:
I am not sure how this works. This seems to me that different parts of this file may have different lifetime.  The newly appended part will have usage of "storage usage" * "this part's lifetime". What if the file's expiry time is earlier than the newly appended part's lifetime and the user doesn't renew or (auto-renew) when the file expires, does this mean the user is over-charged?

Reply:
The lifetime of the newly appended part is exactly the remaining lifetime of the file as it currently exists.

Reply:
So does this mean the FileAppend transaction will change the file's remaining lifetime?